lib/src/git.rs

```diff
@@ -1239,6 +1359,30 @@ pub struct GitFetchStats {
 }
 
 #[tracing::instrument(skip(mut_repo, git_repo, callbacks))]
+pub fn clone(
```

Comment:
nit: `clone()` sounds like it would create new repository. I can't think of a better function name, but it might be `fetch_<something>()` where `<something>` is e.g. `new_remote`.

Reply:
I went with `fetch_and_get_default_branch`, which is a mouthful!

However, I'm going ahead to simplify cli/src/commands/git/clone.rs and cli/src/commands/git/fetch.rs directly in the next commit(s). I'm going to get rid of these `git::fetch*` functions and use the new `git::GitFetch` API directly.